c_reference/tests/kws/test_keyword_spotting.c

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
```

Comment:
- We need some description of the overall architecture, and clarification over the form of input being tested and the nature of output. 
- can this code be extended for streaming data easily?
- The file name could also be more specific as to which model arch for keyword spotting tested here.

Reply:
I've added the description and a note on what to cater to when porting the code to the microcontroller. 

The code needs minor modifications for handling the streaming data. These changes are mostly variable assignments, for example fixing input streaming buffer it steps, feature length